### ISSUE TITLE
[BUG] Receive error when running kaggle criteo dlrm 

### ISSUE THREAD

**Issue by iidsample** [NVIDIA-Merlin/HugeCTR#301]
**Describe the bug**
On trying to run kaggle criteo dataset using docker container nvcr.io/nvidia/merlin/merlin-training:22.03 i receive a dataset parser error. 
```
RuntimeError: Runtime error: dataset key type and dataset size is not compatible.
        Error_t::UnspecificError at create_drwg_raw(/hugectr/HugeCTR/include/data_readers/data_reader.hpp:301)
```
I think the error is being generated from [here](https://github.com/NVIDIA-Merlin/HugeCTR/blob/master/HugeCTR/include/data_readers/data_reader.hpp#L301)
**To Reproduce**
I have followed exactly the same instructions as provided in the tutorial for running kaggle dataset [here](https://github.com/NVIDIA-Merlin/HugeCTR/tree/master/samples/dlrm)
I guess I will still paste the exact commands - 
1. docker pull nvcr.io/nvidia/merlin/merlin-training:22.03
2. docker run --privileged=true --ipc host --name crtieo_trun --network host --gpus all -it -d -v /hdd2/criteo_terabyte_docker_cp/:/workspace nvcr.io/nvidia/merlin/merlin-training:22.03
3. Download the kaggle dac dataset.
4. tar zxvf kaggle.tar.gz
5. We get two files - train.txt (45840617 lines) and val.txt(6042135 lines)
6. dlrm_raw ./ ./
7. python dlrm_kaggle_fp32.py 

**Expected behavior**
I expected the code to.

**Environment (please complete the following information):**
 - OS:Ubuntu 18.04
 - Graphic card: RTX 2080 TI
 - CUDA version: CUDA 11.6
 - Docker image - merlin-training:22.03

**Comment by minseokl** [comment]
Hi @iidsample 

It appears that the error is from the mismatch between the actual dataset size and expected dataset size.
For instance, the number of samples in your training dataset must be the same as `num_samples = 36634624` as specified in `dlrm_kaggle_fp32.py`. So does the `num_eval_samples`. Could you check how many samples your `train.bin` and `test.bin` have?  You can also check if their sizes are as the same as 5.5GB (train.bin) and 700MB (test.bin)

Thanks,
Minseok

**Comment by iidsample** [comment]
Hi, 

I did try to look at the number of lines. However since this is a binary it doesn't look like it is possible. 
The size does match though

**Comment by gnupdev** [comment]
I also got the same error.

Set the num_samples=36672493, eval_num_samples=4584062.

Label + Dense + Sparse input = 40 * 4 Bytes. (per line in train.txt)

```
root@4bff9b9bd552:/workspace/kaggle# dlrm_raw ./ ./
[HCTR][04:48:26][INFO][RK0][main]: Processing Kaggle datasets
[HCTR][04:48:26][INFO][RK0][main]: input_dir: ./
[HCTR][04:48:26][INFO][RK0][main]: output_dir: ./
[HCTR][04:48:32][INFO][RK0][main]: .//train.txt's total rows number = 45840617
[HCTR][04:48:32][INFO][RK0][main]: Slot size array in .//train.txt, missing value mapped to unused key:
1459, 583, 6373320, 1977439, 305, 24, 12513, 633, 3, 92719, 5681, 5666265, 3193, 27, 14986, 4209368, 10, 5652, 2173, 4, 5058596, 18, 15, 282062, 105, 141594,
[HCTR][04:48:32][INFO][RK0][main]: Time used to build map: 5197 milliseconds.
[HCTR][04:48:32][INFO][RK0][main]: Total bytes read: 11147184845 Effective Read B/W: 1.99762 GB/s.
[HCTR][04:48:42][INFO][RK0][main]: Porcessed file: .//train.txt for /train_data.bin
[HCTR][04:48:42][INFO][RK0][main]: Size of train_data.bin: 5867598880 Bytes.
[HCTR][04:48:48][INFO][RK0][main]: Size of val_data.bin: 733449920 Bytes.
[HCTR][04:48:55][INFO][RK0][main]: Size of test_data.bin: 733449920 Bytes.
[HCTR][04:48:55][INFO][RK0][main]: Processed file: .//train.txt for val_data.bin and test_data.bin
[HCTR][04:48:55][INFO][RK0][main]: Time to process binaries: 22692 milliseconds.
[HCTR][04:48:55][INFO][RK0][main]: Convert Bytes reading: 33441554535, Effective reading B/W: 1.3725 GB/s.
[HCTR][04:48:55][INFO][RK0][main]: Convert Bytes writing: 7334498720, Effective reading B/W: 0.301022 GB/s.
[HCTR][04:48:55][INFO][RK0][main]: Convert Bytes total: 40776053255, Effective reading B/W: 1.67353 GB/s.
[HCTR][04:48:55][INFO][RK0][main]: Application process B/W: 0.371449 GB/s.
[HCTR][04:48:55][INFO][RK0][main]: Done.
```

**Comment by minseokl** [comment]
Thank you @iidsample and @gnupdev 

@jershi425 will take a look the issue based on the information you provided.

**Comment by jershi425** [comment]
Thank you @iidsample and @gnupdev
I was able to reproduce this error. And @gnupdev is right about the correct num_samples. Changing it to 36672493 will solve this issue. We will also update the python script from our side.
Thanks again!

**Comment by zehuanw** [comment]
This issue is just fixed in the v3.6. 

Please find the updated script here: https://github.com/NVIDIA-Merlin/HugeCTR/blob/master/samples/dlrm/dlrm_kaggle_fp32.py here. 

Let me close this issue first. If you still encounter this please reopen it.
